### PR TITLE
ackley, rastrigin desc

### PR DESCRIPTION
**Pull request by milakis**

### CODE REVIEW

File: temfpy/optimization.py [hunk 1]
```diff
@@ -1,20 +1,113 @@
...
+
+    Parameters
+    ----------
+
+    x : array_like
+        Input domain with dimension d. It is usually evaluated on the hypercube
+        :math:`x_i\in [-32.768, 32.768]`, for all :math:`i = 1, \dots, d`.
+
+    float a, b, c
+        Recommended variable values are 20, 0.2, 2π, respectively.
```
Comment (loikein) [inline temfpy/optimization.py:27]: This fix is ugly, but it's the only way I can get it to work the way you want. 

Also, default value should come after the parameter description, as seen in [numpydoc docstring guide](https://numpydoc.readthedocs.io/en/latest/format.html#sections).

**Comment by loikein** [comment]
I have reported the bug when combining multiple variables: [Render error when combining multiple input parameters in docstring · Issue #7780 · sphinx-doc/sphinx](https://github.com/sphinx-doc/sphinx/issues/7780)

The only thing left is to wait…

File: temfpy/optimization.py [hunk 2]
```diff
@@ -1,24 +1,125 @@
...
+
+    Parameters
+    ----------
+
+    x : array_like
+        Input domain with dimension d.
+        It is usually evaluated on the hypercube
+        :math:`x_i\in [-32.768, 32.768]`, for all :math:`i = 1, \dots, d`.
+
+    a : float, optional
```
Comment (peisenha) [inline temfpy/optimization.py:23]: The dots are missing in the end of the three default value information.

File: temfpy/optimization.py [hunk 3]
```diff
@@ -1,24 +1,125 @@
...
+    )
+    rslt -= np.exp(1 / len(x) * np.sum(np.cos(np.multiply(c, x))))
 
     return rslt
 
 
 def rastrigin(x, a=10):
     r"""Rastrigin function.
+
+    :math:`f(x) = a d + \sum_{i=1}^n (x_i^2 -10cos(2\pi x_i))`
```
Comment (peisenha) [inline temfpy/optimization.py:81]: The index of the sum should run to `d`, right? Also, please add a space between the 10 and cos.

File: temfpy/optimization.py [hunk 4]
```diff
@@ -1,24 +1,125 @@
...
+    .. [R1974] Rastrigin, L. A. (1974).
+       Systems of extremal control.
+       Moscow, Russia: Mir.
+
+    Examples
+    --------
+
+    >>> x = [0, 0]
+    >>> y = rastrigin(x)
+    >>> np.testing.assert_almost_equal(y, 0)
```
Comment (peisenha) [inline temfpy/optimization.py:119]: I think we need one blank line here? You have one in the function above.

Comment (peisenha) [reply temfpy/optimization.py:119]: Just checked, it is the other way around. No blank lines in the end.

**Pull request merged**